src/onboarding2/Create/Create.js

```diff
@@ -54,73 +70,135 @@ function useTemplateState() {
     } = loadTemplateState()
 
     if (templateId) {
+      updateTemplateScreen(templateId, templateScreenIndex)
       setTemplateData(templateData)
-      setTemplateId(templateId)
-      setTemplateScreenIndex(templateScreenIndex)
     }
-  }, [])
+  }, [updateTemplateScreen])
 
-  const [template, screens] = useMemo(() => {
-    const template = templates.find(template => template.id === templateId)
-    return [
-      template,
-      (template && template.template && template.template.screens) || [],
-    ]
-  }, [templateId])
+  // Save the template state
+  useEffect(() => {
+    if (template && template.id) {
+      saveTemplateState({
+        templateData,
+        templateId: template.id,
+        templateScreenIndex,
+      })
+    }
+  }, [templateData, template, templateScreenIndex])
 
-  const updateScreenIndex = useCallback(
+  const relativeScreen = useCallback(
     diff => {
       const updatedScreenIndex = templateScreenIndex + diff
 
       // Back to the templates selection
-      if (updatedScreenIndex < 0) {
+      if (updatedScreenIndex < 0 || !template) {
         setTemplateData({})
-        setTemplateId(null)
-        setTemplateScreenIndex(-1)
+        updateTemplateScreen(null, -1)
         return
       }
 
-      setTemplateScreenIndex(screenIndex =>
-        Math.min(screens.length - 1, updatedScreenIndex)
+      updateTemplateScreen(
+        template.id,
+        Math.min(template.screens.length, updatedScreenIndex)
       )
     },
-    [screens, templateScreenIndex]
+    [updateTemplateScreen, templateScreenIndex, template]
   )
 
   const prevScreen = useCallback(() => {
-    updateScreenIndex(-1)
-  }, [updateScreenIndex])
+    relativeScreen(-1)
+  }, [relativeScreen])
 
   const nextScreen = useCallback(
     (templateData = {}) => {
       setTemplateData(templateData)
-      updateScreenIndex(1)
+      relativeScreen(1)
     },
-    [updateScreenIndex]
+    [relativeScreen]
   )
 
-  const setTemplate = useCallback(id => {
-    setTemplateId(id)
-    setTemplateScreenIndex(0)
-  }, [])
-
-  const TemplateScreen = useMemo(
-    () =>
-      (screens[templateScreenIndex] && screens[templateScreenIndex][1]) ||
-      (() => null),
-    [screens, templateScreenIndex]
-  )
+  const TemplateScreen = useMemo(() => {
+    return (
+      (template &&
+        template.screens[templateScreenIndex] &&
+        template.screens[templateScreenIndex][1]) ||
+      (() => null)
+    )
+  }, [template, templateScreenIndex])
 
   return {
     TemplateScreen,
     nextScreen,
     prevScreen,
-    setTemplate,
     setTemplateData,
     template,
     templateData,
     templateScreenIndex,
-    templateScreens: screens,
+    updateTemplateScreen,
```

Comment:
Perhaps we can replace the `updateTemplateScreen()` export swith something like `selectTemplate()` that's just a nice, simple wrapper around `updateTemplateScreen()` (all it would do is remove the ability to specify the page); effectively `handleUseTemplate()` already does this but it feels like something that should belong in the hook instead.